pyrovision/datasets/openfire.py

```diff
@@ -20,179 +19,125 @@ class OpenFire(VisionDataset):
+
+    def download(self) -> None:
+        """Download the OpenFire data if it doesn't exist already."""
+        # download files
+        if self.sample:
+            self.url = self.urls['sample']
+            md5 = self.md5s['sample']
+        else:
+            self.url = self.urls['ds']
+            md5 = self.md5s['ds']
 
-        Args:
-            extract (list<dict>): image extract to download
-            threads (int, optional): number of threads used for parallel downloading
-        """
+        path = os.path.normpath(self.url)
+        self.filename = path.split(os.sep)[-1]
 
-        self._images.mkdir(parents=True, exist_ok=True)
-        # Prepare URL and filenames for multi-processing
-        entries = [(s['url'], s['name']) for s in extract
-                   if not self._images.joinpath(s['name']).is_file()]
-        # Use multiple threads to speed up download
-        if len(entries) > 0:
-            download_urls(entries, self._images, threads=threads)
+        if self._check_exists():
+            return
 
-    def _verify_samples(self, extract):
-        """ Download images from a specific extract
+        os.makedirs(self.raw_folder, exist_ok=True)
 
-        Args:
-            extract (list<dict>): list of samples
-        Returns:
-            valid_samples (list<dict>): list of valid samples
-        """
+        try:
+            download_and_extract_archive(
+                self.url, download_root=self.raw_folder,
+                filename=self.filename,
+                md5=md5
+            )
+        except URLError as error:
+            print(
+                "Failed to download (trying next):\n{}".format(error)
+            )
 
-        valid_samples = []
-        dl_issues, target_issues = 0, 0
-        # Verify samples in extract
-        for sample in extract:
-
-            is_ok = True
-            # Verify image
-            if not self._images.joinpath(sample['name']).is_file():
-                dl_issues += 1
-                is_ok = False
-
-            # Verify targets
-            if self.class_to_idx.get(sample['target']) is None:
-                target_issues += 1
-                is_ok = False
-
-            if is_ok:
-                valid_samples.append(sample)
-
-        # HTTP errors
-        if dl_issues == len(extract):
-            raise RuntimeError('Images not found. You can use download=True to download them.')
-        elif dl_issues > 0:
-            warnings.warn(f'{dl_issues}/{len(extract)} sample images are not present on disk. '
-                          'Please retry downloading later.')
-        # Extract errors
-        if target_issues > 0:
-            warnings.warn(f'{target_issues}/{len(extract)} samples have corrupted targets.')
-
-        return valid_samples
-
-    def extra_repr(self):
+    def extra_repr(self) -> str:
         return "Split: {}".format("Train" if self.train is True else "Test")
```

Comment:
Let's use modern Python3.6 syntax and avoid uppercase for variables
```python
return f"train={self.train}"
```

which will render this `print(dataset)` as
```shell
OpenFire(train=True)
```